assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -144,6 +174,18 @@ const baseResolvers = {
 			yield fetchGetWebDataStreamsStore.actions.fetchGetWebDataStreams( propertyID );
 		}
 	},
+	*getWebDataStreamsBatch( propertyIDs ) {
+		const registry = yield Data.commonActions.getRegistry();
+		const webdatastreams = registry.select( STORE_NAME ).getWebDataStreamsBatch( propertyIDs ) || {};
+
+		const availablePropertyIDs = Object.keys( webdatastreams );
+		const remainingPropertyIDs = difference( propertyIDs, availablePropertyIDs );
+		if ( remainingPropertyIDs.length > 0 ) {
+			for ( let i = 0; i < remainingPropertyIDs.length; i += 10 ) {
+				yield fetchGetWebDataStreamsBatchStore.actions.fetchGetWebDataStreamsBatch( remainingPropertyIDs.slice( i, i + 10 ) );
```

Comment:
This looks like it would work to me but it would be great if we could add a simple test for this, just to make sure nothing is lost in the iteration. Would you please add a test for selecting more than 10 IDs? We don't need to add any additional fixtures for this, we can simply generate 12 or 15 from one of the existing objects. Only the ID really needs to change.

Reply:
Added.